ui/cryptomaterial/segmented_control.go

```diff
@@ -0,0 +1,107 @@
+package cryptomaterial
+
+import (
+	"gioui.org/font"
+	"gioui.org/layout"
+
+	"github.com/crypto-power/cryptopower/ui/values"
+)
+
+type SegmentedControl struct {
+	theme *Theme
+	list  *ClickableList
+
+	selectedIndex int
+	segmentTitles []string
+
+	changed bool
+}
+
+func (t *Theme) SegmentedControl(segmentTitles []string) *SegmentedControl {
+	list := t.NewClickableList(layout.Horizontal)
+	list.IsHoverable = false
+
+	return &SegmentedControl{
+		list:          list,
+		theme:         t,
+		segmentTitles: segmentTitles,
+	}
+}
+
+func (sc *SegmentedControl) Layout(gtx C) D {
+	sc.handleEvents()
+
+	return LinearLayout{
+		Width:      WrapContent,
+		Height:     WrapContent,
+		Background: sc.theme.Color.Background,
+		Border:     Border{Radius: Radius(8)},
+	}.Layout(gtx,
+		layout.Rigid(func(gtx C) D {
+			return sc.list.Layout(gtx, len(sc.segmentTitles), func(gtx C, i int) D {
+				isSelectedSegment := sc.SelectedIndex() == i
+				return layout.Stack{Alignment: layout.Center}.Layout(gtx,
+					layout.Stacked(func(gtx C) D {
+						return layout.Inset{}.Layout(gtx, func(gtx C) D {
+							return layout.Center.Layout(gtx, func(gtx C) D {
```

Comment:
The main widget that displays the button at index i is this last `layout.Center.Layout` line. What's the idea behind placing it inside an empty inset that is stacked inside a stack layout?

Reply:
i had the idea of passing an array of widgets as an argument, to the segmented control widget, the array of widget was meant to be the children of the segmented control, hence the stack layout, i didn't go through with that idea anymore due to the complexity.

i've cleaned up that section of code now